video/out/vo.c

```diff
@@ -489,11 +489,14 @@ static void update_vsync_timing_after_swap(struct vo *vo,
     }
 
     in->num_successive_vsyncs++;
-    if (in->num_successive_vsyncs <= DELAY_VSYNC_SAMPLES)
+    if (in->num_successive_vsyncs <= DELAY_VSYNC_SAMPLES) {
+        in->base_vsync += in->vsync_interval;
```

Comment:
I don't think it is correct to increment `base_vsync` here. It is updated in `update_vsync_timing_after_swap` and this function makes sure valid value is used. base_vsync is last valid point, incrementing with with interval will accumulate a lot of drift, making this value unreliable.

Reply:
I guess it makes more sense to set it to the actual vsync instead of the nominal vsync interval. Updated the patch